libs/hdf-converters/src/xccdf-results-mapper.ts

```diff
@@ -278,142 +299,149 @@ export class XCCDFResultsMapper extends BaseConverter {
+                        description
+                      ) as string
+                  },
+                  title: {path: ['title.text', 'title']}
                 }
-              },
+              ],
               rule_result: {
-                path: '$.Benchmark.TestResult',
-                transformer: (testResult: Record<string, unknown>): unknown =>
-                  getRuleResultItem(testResult, RULE_RESULT_PATHS)
+                path: ['ruleResult']
               },
               value: {
-                path: '$.Benchmark.Value',
-                transformer: (values: Record<string, unknown>[]): unknown => {
-                  return _.find(values, (value: Record<string, unknown>) => {
-                    const id = _.get(value, 'id');
-                    return id && id === valueIdTracker;
-                  });
-                }
+                path: ['values']
               }
             },
             refs: [],
             source_location: {},
-            title: {path: ['Rule.title.text', 'Rule.title']},
-            id: {
-              path: 'Rule',
-              transformer: (input: Record<string, unknown>): string => {
-                const valueIdPaths = ['check.check-export.value-id'];
-                let setValueIdTracker = false;
-                for (const path of valueIdPaths) {
-                  const valueId = _.get(input, path);
-                  if (valueId !== undefined) {
-                    valueIdTracker = valueId as string; // NOTE: global variable
-                    setValueIdTracker = true;
-                  }
-                }
-                if (!setValueIdTracker) {
-                  valueIdTracker = undefined;
-                }
-                const id = _.get(input, 'id');
-                if (typeof id === 'string') {
-                  idTracker = id; // NOTE: global variable
-                  return id.split('_S')[1].split('r')[0];
-                } else {
-                  return '';
-                }
-              }
-            },
+            title: {path: ['title.text', 'title']},
+            id: {path: ['id']},
             desc: {
-              path: ['Rule.description.text', 'Rule.description'],
-              transformer: (description: string): string => {
-                const descTextJson = convertEncodedXmlIntoJson(description);
-                return _.get(descTextJson, 'VulnDiscussion', '') as string;
-              }
+              path: ['description.text', 'description'],
+              transformer: (description: string): string =>
+                _.get(
+                  parseXml(description),
+                  'ProfileDescription',
+                  description
+                ) as string
             },
             descriptions: [
               {
                 data: {
-                  path: 'Rule.check.check-content-ref.name',
+                  path: ['check.check-content-ref.name'],
```

Comment:
<img width="1846" alt="image" src="https://github.com/mitre/heimdall2/assets/2015821/765799d1-06bd-4b01-baae-12648e07590f">

check text and references are falling back to the correspondingly named tags when they are left as empty string / empty array.  this causes [Object] to show up in heimdall when looking at the details tab.

please update descriptions.check and the refs array to contain this information instead.  refs array will probably be pretty easy - just take the data that looks like this (
![image](https://github.com/mitre/heimdall2/assets/2015821/26ccbd97-fe3c-4dc0-aabb-26663f4a1548)
) and put each one of those objects as that 'ref' attribute (https://github.com/mitre/heimdall2/blob/master/libs/inspecjs/src/generated_parsers/v_1_0/exec-json.ts#L245).  check text will be harder (
![image](https://github.com/mitre/heimdall2/assets/2015821/91bb9f8c-b6e1-4fbb-946c-67d344aceb81)
) - easiest thing to do that i'm fine with is just running a JSON.stringify and dumping it out.

Reply:
whereas the ref/reference stuff is not overlapped, the check subdescription and check tag do have overlap where if the subdescription is available it hides the tag.  for now, we have stringified the tag so it looks good in heimdall, but we should go and make the subdescription find things properly and behave well with the original data being an array for example.